sklearn/datasets/_openml.py

```diff
@@ -601,6 +688,7 @@ def fetch_openml(
     as_frame: Union[str, bool] = "auto",
     n_retries: int = 3,
     delay: float = 1.0,
+    parser: Optional[str] = "deprecated",
```

Comment:
We are changing defaults here, so based on [contributing docs](https://scikit-learn.org/dev/developers/contributing.html#change-the-default-value-of-a-parameter), this hsould be "warn"?